### PR TITLE
CI: test pypy

### LINKED ISSUE #1073: Include a CI run with PyPy
This issue is part of the improvements to the CI because we want to move from Appveyor to GitHub actions. One of the improvements includes adding a CI run with PyPy. This includes checking the dependencies in requirements-dev.txt.

### PR DESCRIPTION
**Pull request by fangchenli**
Closes #1073 #1085

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -11,10 +10,7 @@ numpy
 scipy
 selenium
 pytest-xdist
-pytest-rerunfailures>=8.0; platform_python_implementation != "PyPy"
-pytest-rerunfailures>=8.0,<9.0; platform_python_implementation == "PyPy"
-pytest-faulthandler; platform_python_implementation != "PyPy"
-pytest-faulthandler<2.0; platform_python_implementation == "PyPy"
```
Comment (fangchenli) [inline requirements-dev.txt:17]: `pytest-faulthandler` is included in pytest>=5.0

File: requirements-dev.txt [hunk 2]
```diff
@@ -11,10 +10,7 @@ numpy
...
+pytest-rerunfailures>=8.0
```
Comment (datapythonista) [inline requirements-dev.txt:13]: From the pins, seems like 9.0 was broken for pypy. Not sure if whatever problem happened is now fixed or when it was fixed, but if we pin, I'd go for `>=10`, no?

File: test/test_environment.py [hunk 3]
```diff
@@ -586,7 +586,7 @@ def test_pypy_virtualenv(tmpdir):
     for env in environments:
         env.create()
         output = env.run(['-c', 'import sys; print(sys.pypy_version_info)'])
-        assert output.startswith(str("(major="))
+        assert "(major=" in output
```
Comment (datapythonista) [inline test/test_environment.py:589]: There was this PR for this error: #1002 I was waiting to have pypy tests in the CI to merge, happy to get merge there now that this has been tested, or to get the fix here.

CC: @QuLogic

**Comment by datapythonista** [comment]
Thanks @fangchenli, really nice.

**Pull request merged**